compatibility-tests/compile-fail/tests/ui/duplication-backtrace-delegate.stderr

```diff
@@ -1,11 +1,12 @@
-error: Cannot have `backtrace` and `backtrace(delegate)` fields in the same error variant
+error: Cannot have `backtrace` field and `backtrace` attribute on a source field in the same error variant
+  --> $DIR/duplication-backtrace-delegate.rs:10:9
+   |
+10 | /         #[snafu(backtrace)]
+11 | |         source: String,
+   | |______________________^
```

Comment:
This highlight seems a bit odd; are we spanning more than we should?

Reply:
Ah, yes, I found it.  It's because for fields named "source" or "backtrace" I was getting the location using `syn_field`, which represents the entire thing with attributes.

In this specific example, I kind of like the longer span, because it makes the "backtrace attribute on a source field" idea clear.  However, it'd be less clear if there were more attributes on the field - they'd also be highlighted.

I'm not sure the best thing to point at, though, if we wanted to restrict it.  "backtrace attribute on a source field" has those two inner concepts that are both important.  I'd guess the attribute is more important.  But, then, I'm not sure how to get the location of the backtrace attribute on the source field at the time when we're checking whether we saw a source attribute.  Either I'd store the backtrace location as another awkward piece of data in the source attribute tracking, or dig back through the attributes on the field, or...

I think the larger span is actually not a bad compromise...

Reply:
I don't think we have the full procedural macro power we need with the current version of Rust. The compiler proper has the ability to do multiple spans; I'd love it if it could be

```
cannot have backtrace attribute on source field when there is already a backtrace field
NN #[snafu(backtrace)]
   ^~~~~~~~~~~~~ attribute declared here
NN source: Box<Error>,
    ^~~~~~~~~ field declared here
```

Reply:
If that's not possible, then I'm OK with the longer span. Reporting the error is the backwards compatibility issue, improving the span (if possible) should be able to happen later.

Reply:
There was https://github.com/dtolnay/syn/issues/558 requesting errors with multiple spans in syn, which was closed in preference to it being built in another library.  It seems the [proc-macro Diagnostic](https://doc.rust-lang.org/stable/proc_macro/struct.Diagnostic.html) API might be the way of the future, once it's implemented and stabilized in https://github.com/rust-lang/rust/issues/54140.  I agree that it doesn't seem like we have a convenient way currently.